ui/contexts/gasFee.js

```diff
@@ -31,7 +31,7 @@ export function useGasFeeContext() {
 GasFeeContextProvider.propTypes = {
   children: PropTypes.node.isRequired,
   defaultEstimateToUse: PropTypes.string,
-  transaction: PropTypes.object.isRequired,
+  transaction: PropTypes.object,
```

Comment:
Interesting, is there a reason why you wouldn't have a transaction when requesting gas fee estimates?

Reply:
Yep I have been thinking of doing it from some time, it is coz of this condition: https://github.com/MetaMask/metamask-extension/blob/develop/ui/hooks/gasFeeInput/useGasFeeInputs.js#L99

Having transaction as required will prevent testing of this condition.

Though in reality I can not think of a scenario when transaction will be undefined.